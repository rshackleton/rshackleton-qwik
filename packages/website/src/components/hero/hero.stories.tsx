import type { Meta, StoryObj } from 'storybook-framework-qwik';
import type { HeroProps } from './hero';
import { Hero } from './hero';

export default {
  title: 'Hero',
  component: Hero,
} as Meta<HeroProps>;

export const Default: StoryObj<HeroProps> = {};
