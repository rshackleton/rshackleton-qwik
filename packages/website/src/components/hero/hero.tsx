import { component$ } from '@builder.io/qwik';
import { Link } from '@builder.io/qwik-city';

export type HeroProps = {};

export const Hero = component$<HeroProps>(() => {
  return (
    <section class="relative grid place-content-center place-items-center overflow-hidden py-[8vw]">
      <div class="relative mx-auto w-full max-w-screen-2xl px-4 text-center">
        <h1 class="pb-5 text-5xl font-bold">Full Stack Developer</h1>
        <p class="mx-auto max-w-prose pb-8 text-lg">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
          ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
          ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </p>
        <Link
          class="bg-primary-500 hover:bg-primary-400 inline-block scale-100 rounded py-3 px-5 text-base font-bold text-white subpixel-antialiased transition will-change-transform hover:scale-110"
          href="/about"
        >
          Find out more
        </Link>
      </div>
    </section>
  );
});
