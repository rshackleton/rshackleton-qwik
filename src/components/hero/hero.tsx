import { component$ } from '@builder.io/qwik';

export const Hero = component$(() => {
  return (
    <section>
      <div class="site-container py-4">
        <h1>Hello World</h1>
      </div>
    </section>
  );
});
