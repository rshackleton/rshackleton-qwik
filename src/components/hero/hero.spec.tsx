import { createDOM } from '@builder.io/qwik/testing';
import { expect, test } from 'vitest';
import { Hero } from './hero';

test(`[Hero Component]: Renders without error`, async () => {
  const { screen, render } = await createDOM();
  await render(<Hero />);
  expect(screen.outerHTML).toMatchSnapshot();
});
