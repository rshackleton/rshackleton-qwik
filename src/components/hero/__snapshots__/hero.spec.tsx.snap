// Vitest Snapshot v1

exports[`[Hero Component]: Renders without error 1`] = `"<host q:version=\\"0.16.2\\" q:container=\\"resumed\\" q:render=\\"dom-dev\\"><!--qv q:key=bfJE:eC_0 data-qrl=Hero_component_bfJE09oKJq8--><section><div class=\\"site-container py-4\\"><h1>Hello World</h1></div></section><!--/qv--></host>"`;
